### PR TITLE
ROX-23445: Adding namespace exclusions to default policies

### PR DESCRIPTION
**Pull request by ksurabhi91**
## Description

A detailed explanation of the changes in your PR.

Feel free to remove this section if it is overkill for your PR, and the title of your PR is sufficiently descriptive.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

TODO(replace-me)  
Use this space to explain **how you validated** that **your change functions exactly how you expect it**.
Feel free to attach JSON snippets, curl commands, screenshots, etc. Apply a simple benchmark: would the information you
provided convince any reviewer or any external reader that you did enough to validate your change.

It is acceptable to assume trust and keep this section light, e.g. as a bullet-point list.

It is acceptable to skip testing in cases when CI is sufficient, or it's a markdown or code comment change only.  
It is also acceptable to skip testing for changes that are too taxing to test before merging. In such case you are
responsible for the change after it gets merged which includes reverting, fixing, etc. Make sure you validate the change
ASAP after it gets merged or explain in PR when the validation will be performed.  
Explain here why you skipped testing in case you did so.

Have you created automated tests for your change? Explain here which validation activities you did manually and why so.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

**Comment by clickboo** [comment]
Any chance we can fold this into the 199->200 migration? cc: @dashrews78

File: migrator/migrations/m_200_to_m_201_openshift_policy_exclusions/migration.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
+package m200tom201
```
Comment (dashrews78) [inline migrator/migrations/m_200_to_m_201_openshift_policy_exclusions/migration.go:1]: 200-201 is the compliance migration.  So you need to renumber this one.

**Review by dashrews78 — DISMISSED** [review]
You need to use the bootstrap to start the migration.  
200 is already taken by compliance.  This one would need to be 201 and change all the appropriate version files and such

File: migrator/migrations/m_201_to_m_202_openshift_policy_exclusions/migration.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+	"github.com/stackrox/rox/migrator/migrations/m_201_to_m_202_openshift_policy_exclusions/schema"
+	"github.com/stackrox/rox/migrator/migrations/policymigrationhelper"
+	"github.com/stackrox/rox/migrator/types"
+	"gorm.io/gorm"
+)
+
+var (
+	migration = types.Migration{
+		StartingSeqNum: 200,
+		VersionAfter:   &storage.Version{SeqNum: 200},
```
Comment (clickboo) [inline migrator/migrations/m_201_to_m_202_openshift_policy_exclusions/migration.go:19]: 201?

File: migrator/migrations/m_201_to_m_202_openshift_policy_exclusions/migration.go [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+	policyDiffFS embed.FS
+
+	// We want to migrate only if the existing policy sections,name and description haven't changed.
+	fieldsToCompare = []policymigrationhelper.FieldComparator{
+		policymigrationhelper.DescriptionComparator,
+		policymigrationhelper.PolicySectionComparator,
+		policymigrationhelper.NameComparator,
+	}
+
+	policyDiffs = []policymigrationhelper.PolicyDiff{
```
Comment (clickboo) [inline migrator/migrations/m_201_to_m_202_openshift_policy_exclusions/migration.go:39]: secret_env and sensitive_files missing in this list

File: migrator/migrations/m_203_to_m_204_openshift_policy_exclusions/migration.go [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+	"github.com/stackrox/rox/migrator/migrations/m_203_to_m_204_openshift_policy_exclusions/schema"
+	"github.com/stackrox/rox/migrator/migrations/policymigrationhelper"
+	"github.com/stackrox/rox/migrator/types"
+	"gorm.io/gorm"
+)
+
+var (
+	migration = types.Migration{
+		StartingSeqNum: 203,
+		VersionAfter:   &storage.Version{SeqNum: 203},
```
Comment (dashrews78) [inline migrator/migrations/m_203_to_m_204_openshift_policy_exclusions/migration.go:19]: 204

**Comment by dashrews78** [comment]
@ksurabhi91 there are a few things missing that are often generated.  I recommend you start over and run the bootstrap make target mentioned in the migrator README.  Then copy all your changes over into there.  That bootstrap will automatically take care of the sequence and registering the migration and the boiler plate stuff.  Reduces the risk of missing something.

**Comment by dashrews78** [comment]
@ksurabhi91 You need to update `migrator/runner/all.go` which the bootstrap does.

File: pkg/defaults/policies/files/exec-iptables-root.json [hunk 5]
```diff
@@ -13,6 +13,34 @@
...
+    {
+      "name": "Don't alert on deployment keepalived-worker-* in namespace openshift-kni-infra",
+      "deployment": {
+        "name": "keepalived-worker-*",
+        "scope": {
+          "namespace": "openshift-kni-infra"
+        }
+      }
+    }
+  ,
```
Comment (dashrews78) [inline pkg/defaults/policies/files/exec-iptables-root.json:43]: This is one of your lint errors and would need to be put in the migration/after as well.

```suggestion
    },
```

File: migrator/migrations/m_203_to_m_204_openshift_policy_exclusions_for_4_5/migration.go [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+	policyDiffFS embed.FS
+
+	// We want to migrate only if the existing policy sections,name and description haven't changed.
+	fieldsToCompare = []policymigrationhelper.FieldComparator{
+		policymigrationhelper.DescriptionComparator,
+		policymigrationhelper.PolicySectionComparator,
+		policymigrationhelper.NameComparator,
+	}
+
+	policyDiffs = []policymigrationhelper.PolicyDiff{
```
Comment (dashrews78) [inline migrator/migrations/m_203_to_m_204_openshift_policy_exclusions_for_4_5/migration.go:43]: nit:  order these alphabetically by filename to make it easier to compare to the file listings.

**Review by clickboo — APPROVED** [review]
🚢 after CI 🟢

**Pull request merged**